src/jshint.js

```diff
@@ -4584,7 +4605,25 @@ var JSHINT = (function() {
   FutureReservedWord("public", { es5: true, strictOnly: true });
   FutureReservedWord("short");
   FutureReservedWord("static", { es5: true, strictOnly: true });
-  FutureReservedWord("super", { es5: true });
+  FutureReservedWord("super", {
+    es5: true,
+    nud: function() {
+      if (state.inESNext() && checkPunctuators(state.tokens.next, "(")) {
+        var f = state.funct;
+        while (f) {
+          // If `super` is called in an arrow function, check the outer scope
+          if (!f["(arrow)"]) { break; }
+          f = f["(context)"];
+        }
+        if (!f["(super)"]) {
+          error("E061", state.tokens.curr);
+        } else {
+          f["(super)"] = "called";
```

Comment:
Looking in the spec it seems it is allowed (and [babel doesn't warn](http://babeljs.io/repl/#?experimental=false&evaluate=true&loose=true&spec=false&playground=false&code=class%20A%20extends%20Array%20%7B%0A%20%20constructor()%20%7B%0A%20%20%20%20super()%3B%0A%20%20%20%20super()%3B%0A%20%20%7D%0A%7D))